content/Improve your Rails App searches with Elasticsearch - Leandro Dominguez.md

```diff
@@ -0,0 +1,213 @@
+
+And last but not least, we are declaring `topic` as a _keyword_, meaning we are expecting this field only to be tokenized (and not processed or analyzed) making the inserts and edits faster. This is because we are not expected to search through this field, but feel free to play with different options.
+
+**Step 4: Starting our Elasticsearch instance and creating our index**
+
+Assuming you already installed Elasticsearch, start an instance of Elastic by just executing in a terminal **elasticsearch** (remember I’m using _homebrew_). This will launch a server in the 9200 port by default. To check if everything is working as expected make a get request to [http://localhost:9200/](http://localhost:9200/) (could be just opening the URL in a browser).
+
+The response should see something like this:
+
+```json
+{
+  "name" : "192.168.0.100",
+  "cluster_name" : "elasticsearch_instance",
+  "cluster_uuid" : "LjE5yneZRWOUFnlasU6y7g",
+  "version" : {
+    "number" : "7.4.2",
+    "build_flavor" : "default",
+    "build_type" : "tar",
+    "build_hash" : "2f90bbf7b93631e52bafb59b3b049cb44ec25e96",
+    "build_date" : "2019-10-28T20:40:44.881551Z",
+    "build_snapshot" : false,
+    "lucene_version" : "8.2.0",
+    "minimum_wire_compatibility_version" : "6.8.0",
+    "minimum_index_compatibility_version" : "6.0.0-beta1"
+  },
+  "tagline" : "You Know, for Search"
+}
+```
+
+Now we have our Elastic instance up and running, open a Rails console and create the corresponding index by executing:
+
+```ruby
+Post.__elasticsearch__.create_index!
+```
+
+If you already had some posts created, you will need to reindex them by running the following:
+
+```ruby
+Post.import
+```
+
+This can take A LOT of time if you have too many records, since Elastic has to analyze and tokenize all of them.
+
+**Step 5: define the search method we will call from the controller**
+
+In the Posts’ model define this basic method that will execute a query in our Elasticsearch instance, taking a string and looking in the specified fields.
+
+```ruby
+   def self.our_first_query(query)
+     search(
+       query: {
+         multi_match: {
+           query: query,
+           fields: %w[title body^5]
+         }
+       }
+     )
+   end
+```
+
+This simple query will search through all our posts and retrieve them in order of importance. The ‘`^5`’ after `body` will make a match in this attribute to weigh 5 more times than one on `title`.
```

Comment:
Wouldn't it make more sense the other way around?

Reply:
I think both works and makes sense in my mind, but I'll do what you're suggesting!

Reply:
Mmmh, I think both make sense. In my case, I intentionally wanted to give more importance to the body, since there was the actual information. Although I understand your suggestion, I think I'll keep it like this so I don't have to change all search examples.

Reply:
Cool, agreed